src/models/users/remote.ts

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+export class RemoteUser {
+
+    /**
+     * @param identifier The unique ID for this user.
+     * @param data The serialized key-value data object to assign to this user.
+     * @throws If identifier is not supplied.
+     */
+    constructor (public readonly id: string, public data: Record<string, unknown>) {
```

Comment:
Why is data not readonly? This allows calls from the the outside to redefine it.